aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -215,10 +212,7 @@ struct cpu_scatter_gather_base_kernel {
     dim = maybe_wrap_dim(dim, self.dim());
 
     scatter_gather_dtype_check(method_name, self, index, src);
```

Comment:
It feels like you should be able to dispense with the dtype checks now too, as they're done in the meta function

Reply:
The problem is that (as mentioned above) `gather` uses it. If we make `gather` structured, it can go away, though.